config/config.go

```diff
@@ -225,6 +243,20 @@ func (c *Config) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	return checkOverflow(c.XXX, "config")
 }
 
+func smtpAuthParam(key string, gc *GlobalConfig) (string, Secret) {
+	// Try to pull it from the global config
+	globalKey := "SMTPAuth" + key
+	reflected := reflect.Indirect(reflect.ValueOf(gc))
```

Comment:
This is getting a bit magic, I think it'd be okay to just switch to config file for this.